### ISSUE TITLE
fix methods that return Task to return Task.FromResult(0) instead of null

### ISSUE THREAD

**Issue by mgmccarthy** [HTBox/allReady#613]
methods that return a Task should not return null:
```
Task IAllReadyDataAccess.DeleteTaskSignupAsync(int taskSignupId)
{
    var taskSignup = _dbContext.TaskSignups.SingleOrDefault(c => c.Id == taskSignupId);

    if (taskSignup != null)
    {
        _dbContext.TaskSignups.Remove(taskSignup);
        return _dbContext.SaveChangesAsync();
    }
    return null;
}
```

Instead of returning `null`, we should be returning `Task.FromResult(0)`

**Comment by mgmccarthy** [comment]
methods that return a Task should not return null:
```
Task IAllReadyDataAccess.DeleteTaskSignupAsync(int taskSignupId)
{
    var taskSignup = _dbContext.TaskSignups.SingleOrDefault(c => c.Id == taskSignupId);

    if (taskSignup != null)
    {
        _dbContext.TaskSignups.Remove(taskSignup);
        return _dbContext.SaveChangesAsync();
    }
    return null;
}
```

Instead of returning `null`, we should be returning `Task.FromResult(0)`

**Comment by YannickMeeus** [comment]
I've picked this up, PR incoming.

**Comment by tonysurma** [comment]
closed in PR #614
